lib/openfeature/sdk/evaluation/context.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module OpenFeature
+  module SDK
+    module Evaluation
+      # A container for arbitrary contextual data that can be used as a basis for dynamic evaluation
+      class Context < SimpleDelegator
+
+        def initialize(context = Concurrent::Hash.new({}))
+          raise ArgumentError, "context must be a Hash" unless context.is_a?(Hash)
+
+          context = Concurrent::Hash[context] unless context.is_a?(Concurrent::Hash)
+          super(context)
+        end
+
+        def freeze
+          super
+          deep_freeze
+        end
+
+        def targeting_key
+          self[:targeting_key]
+        end
+
+        def targeting_key=(value)
+          raise ArgumentError, "targeting_key must be a String" unless value.is_a?(String)
+
+          self[:targeting_key] = value
+        end
```

Comment:
What is `targetting key` refer to exactly?

Reply:
The targeting key is basically a well-defined property in the context, which is necessary because many providers _need_ a user-identifier or equivalent of some kind: https://openfeature.dev/specification/sections/evaluation-context#requirement-311